fabric-resource-loader-v0/src/main/java/net/fabricmc/fabric/impl/resource/loader/ModResourcePackUtil.java

```diff
@@ -67,33 +78,85 @@ public static void appendModResourcePacks(List<ModResourcePack> packs, ResourceT
 				continue;
 			}
 
-			ModResourcePack pack = ModNioResourcePack.create(container.getMetadata().getId(), container, subPath, type, ResourcePackActivationType.ALWAYS_ENABLED);
+			ModResourcePack pack = ModNioResourcePack.create(container.getMetadata().getId(), container, subPath, type, ResourcePackActivationType.ALWAYS_ENABLED, true);
 
 			if (pack != null) {
 				packs.add(pack);
 			}
 		}
 	}
 
-	public static boolean containsDefault(ModMetadata info, String filename) {
-		return "pack.mcmeta".equals(filename);
+	public static void refreshAutoEnabledPacks(List<ResourcePackProfile> enabledProfiles, Map<String, ResourcePackProfile> allProfiles) {
+		LOGGER.debug("[Fabric] Starting internal pack sorting with: {}", enabledProfiles.stream().map(ResourcePackProfile::getName).toList());
+		enabledProfiles.removeIf(profile -> ((FabricResourcePackProfile) profile).isHidden());
+		LOGGER.debug("[Fabric] Removed all internal packs, result: {}", enabledProfiles.stream().map(ResourcePackProfile::getName).toList());
+		ListIterator<ResourcePackProfile> it = enabledProfiles.listIterator();
+		// LinkedHashSet for debug log sort, will switch to normal HashSet once finished
```

Comment:
Does this need doing?

Reply:
Will just remove the comment, since we keep the debug? Shouldn't hurt perf that much, as the injection target is IO-intensive anyway.